server/apps/publish/archive_publish.py

```diff
@@ -79,8 +79,10 @@ def update(self, id, updates, original):
             user = get_user()
             push_notification('item:publish', item=str(item.get('_id')), user=str(user))
             return item
-        except KeyError:
-            raise SuperdeskApiError.badRequestError(message="A non-existent content id is requested to publish")
+        except KeyError as e:
+            raise SuperdeskApiError.badRequestError(
+                message="Key is missing on article to be published: {}"
+                .format(str(e)))
```

Comment:
Probably good to suppress the exception context here ( raise ... **`from None`**) to avoid too verbose stack traces? After all we are only interested in the last error (which is already descriptive enough) and not in the original KeyError.